Sources/SwiftUINavigationCore/NavigationDestination+Vanilla.swift

```diff
@@ -0,0 +1,79 @@
+#if canImport(SwiftUI)
+  import SwiftUI
+
+  @available(iOS, introduced: 16, obsoleted: 17)
+  @available(macOS, introduced: 13, obsoleted: 14)
+  @available(tvOS, introduced: 16, obsoleted: 17)
+  @available(watchOS, introduced: 9, obsoleted: 10)
+  extension View {
+    /// A backport of SwiftUI's `navigationDestination(item:)`.
+    ///
+    /// > Important: This modifier is a wrapper around `navigationDestination(isPresented:)`, which
+    /// > is known to be buggy (especially pre-iOS 16.4, etc.). It is not guaranteed, nor is it
+    /// > expected, to behave exactly like SwiftUI's built-in `navigationDestination(item:)`
+    /// > modifier. Be sure to test both this backport and the native modifier in your application
+    /// > to ensure acceptable behavior.
```

Comment:
We shouldn't be in the business of promising support for these modifiers. We should claim that we are doing just the minimal amount of work to call Apple's own APIs under the hood.

Still I did include the bind workaround in this overload since we already use it in the `unwrapping:` version, and know `navigationDestination` breaks quickly otherwise in iOS <16.4...but maybe we shouldn't even attempt to do that...